### ISSUE TITLE
Referencing subfigures doesn't work as expected

### ISSUE THREAD

**Issue by codingS3b** [MIDL-Conference/MIDLLatexTemplate#19]
When trying to reference subfigures, in addition to the correct name of the subfigure, also the word `subfigure` is printed which is weird.

![image](https://user-images.githubusercontent.com/31093529/159008298-280166d2-724d-4655-88ff-e54f4150849b.png)

Could you propose a way to solve this? Or am I using the provided commands in a wrong way?

An example to reproduce the issue can be found here:

```tex
\documentclass{midl} % Include author names
%\documentclass[anon]{midl} % Anonymized submission

% The following packages will be automatically loaded:
% jmlr, amsmath, amssymb, natbib, graphicx, url, algorithm2e
% ifoddpage, relsize and probably more
% make sure they are installed with your latex distribution

\usepackage{mwe} % to get dummy images

% Header for extended abstracts
\jmlrproceedings{MIDL}{Medical Imaging with Deep Learning}
\jmlrpages{}
\jmlryear{2021}

% to be uncommented for submissions under review
\jmlrworkshop{Short Paper -- MIDL 2022 submission}
\jmlrvolume{-- Under Review}
\editors{Under Review for MIDL 2022}

\title[Short Title]{Full Title of Article}

 % Use \Name{Author Name} to specify the name.
 % If the surname contains spaces, enclose the surname
 % in braces, e.g. \Name{John {Smith Jones}} similarly
 % if the name has a "von" part, e.g \Name{Jane {de Winter}}.
 % If the first letter in the forenames is a diacritic
 % enclose the diacritic in braces, e.g. \Name{{\'E}louise Smith}

 % Two authors with the same address
 % \midlauthor{\Name{Author Name1} \Email{abc@sample.edu}\and
 %  \Name{Author Name2} \Email{xyz@sample.edu}\\
 %  \addr Address}

 % Three or more authors with the same address:
 % \midlauthor{\Name{Author Name1} \Email{an1@sample.edu}\\
 %  \Name{Author Name2} \Email{an2@sample.edu}\\
 %  \Name{Author Name3} \Email{an3@sample.edu}\\
 %  \addr Address}


% Authors with different addresses:
% \midlauthor{\Name{Author Name1} \Email{abc@sample.edu}\\
% \addr Address 1
% \AND
% \Name{Author Name2} \Email{xyz@sample.edu}\\
% \addr Address 2
% }

%\footnotetext[1]{Contributed equally}

% More complicate cases, e.g. with dual affiliations and joint authorship
\midlauthor{\Name{Author Name1\midljointauthortext{Contributed equally}\nametag{$^{1,2}$}} \Email{abc@sample.edu}\\
\addr $^{1}$ Address 1 \\
\addr $^{2}$ Address 2 \AND
\Name{Author Name2\midlotherjointauthor\nametag{$^{1}$}} \Email{xyz@sample.edu}\\
\Name{Author Name3\nametag{$^{2}$}} \Email{alphabeta@example.edu}\\
\Name{Author Name4\midljointauthortext{Contributed equally}\nametag{$^{3}$}} \Email{uvw@foo.ac.uk}\\
\addr $^{3}$ Address 3 \AND
\Name{Author Name5\midlotherjointauthor\nametag{$^{4}$}} \Email{fgh@bar.com}\\
\addr $^{4}$ Address 4
}

\begin{document}

\maketitle

\begin{abstract}
This is a great paper and it has a concise abstract.
\end{abstract}

\begin{keywords}
List of keywords, comma separated.
\end{keywords}

\section{Introduction}

% This is where the content of your paper goes.  Some random notes:
% \begin{itemize}
% \item You should use \LaTeX \cite{Lamport:Book:1989}.
% \item JMLR/PMLR uses natbib for references. For simplicity, here, \verb|\cite|  defaults to
%   parenthetical citations, i.e. \verb|\citep|. You can of course also
%   use \verb|\citet| for textual citations.
% \item You should follow the guidelines provided by the conference.
% \item Read through the JMLR template documentation for specific \LaTeX
%   usage questions.
Note that the JMLR template provides many handy functionalities
such as \verb|\figureref| to refer to a figure,
e.g. \figureref{fig:example},  \verb|\tableref| to refer to a table,
e.g. \tableref{tab:example} and \verb|\equationref| to refer to an equation,
e.g. \equationref{eq:example}.

We want to refer to the overall \figureref{fig:example2} but also to \figureref{fig:2a} and \figureref{fig:2b} as well.

\begin{table}[htbp]
 % The first argument is the label.
 % The caption goes in the second argument, and the table contents
 % go in the third argument.
\floatconts
  {tab:example}%
  {\caption{An Example Table}}%
  {\begin{tabular}{ll}
  \bfseries Dataset & \bfseries Result\\
  Data1 & 0.12345\\
  Data2 & 0.67890\\
  Data3 & 0.54321\\
  Data4 & 0.09876
  \end{tabular}}
\end{table}

\begin{figure}[htbp]
 % Caption and label go in the first argument and the figure contents
 % go in the second argument
\floatconts
  {fig:example}
  {\caption{Example Image}}
  {\includegraphics[width=0.5\linewidth]{example-image}}
\end{figure}

\begin{figure}[htbp]
	\floatconts
	{fig:example2} % label for whole figure
	{\caption{Here comes text.}}% caption for whole figure
	{
		\begin{minipage}[c]{.45\textwidth}
			\subfigure[part 1.][c]{%
				\label{fig:2a}% label for this sub-figure
				\includegraphics[width=0.5\linewidth]{example-image}
			}
		\end{minipage}
		\begin{minipage}[c]{0.45\textwidth}
			\subfigure[part 2.][c]{
				%
				\label{fig:2b}% label for this sub-figure
				\includegraphics[width=0.5\linewidth]{example-image}
			}
			\\
			\\
			\subfigure[part 3.][c]{%
				\label{fig:2c}% label for this sub-figure
				\includegraphics[width=0.5\linewidth]{example-image}
			}
		\end{minipage}
	}
\end{figure}



% \begin{algorithm2e}
% \caption{Computing Net Activation}
% \label{alg:net}
%  % older versions of algorithm2e have \dontprintsemicolon instead
%  % of the following:
%  %\DontPrintSemicolon
%  % older versions of algorithm2e have \linesnumbered instead of the
%  % following:
%  %\LinesNumbered
% \KwIn{$x_1, \ldots, x_n, w_1, \ldots, w_n$}
% \KwOut{$y$, the net activation}
% $y\leftarrow 0$\;
% \For{$i\leftarrow 1$ \KwTo $n$}{
%   $y \leftarrow y + w_i*x_i$\;
% }
% \end{algorithm2e}

% Acknowledgments---Will not appear in anonymized version
\midlacknowledgments{We thank a bunch of people.}

\bibliography{midl-samplebibliography}

\end{document}

```

**Comment by tvercaut** [comment]
The main jmlr documentation has material on sub-figures:
http://mirrors.ctan.org/macros/latex/contrib/jmlr/jmlr.pdf

Can you try the approach suggested there?

**Comment by codingS3b** [comment]
Actually I was following along the sections of the pdf you posted, @tvercaut. My code for the figure is pretty much the same as the example on page 15 in the documentation (except that I am using minpages as well). Also, page 20 of the documentation states to use the `\figureref` command to cross-reference subfigures which I also did in the example.

Did I overlook other parts of the documentation that you had in mind?

**Comment by tvercaut** [comment]
@codingS3b: I can't reproduce your issue but I see that a related bug was fixed upstream in jmlr v1.27:
https://www.dickimaw-books.com/bugtracker.php?key=157

Can you try updating the version? It it solves it I would suggest closing this issue and not implementing a work-around in our classes.

**Comment by tvercaut** [comment]
Closing due to not being able to replicate and lack of activity on this issue.
